pandas/core/reshape/pivot.py

```diff
@@ -507,17 +511,30 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
         df = DataFrame(data)
         df['__dummy__'] = 0
         table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
-                               aggfunc=len, margins=margins,
-                               margins_name=margins_name, dropna=dropna)
+                               aggfunc=len, margins=False, dropna=dropna)
         table = table.fillna(0).astype(np.int64)
 
     else:
         data['__dummy__'] = values
         df = DataFrame(data)
         table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
-                               aggfunc=aggfunc, margins=margins,
-                               margins_name=margins_name, dropna=dropna)
+                               aggfunc=aggfunc, margins=False, dropna=dropna)
 
+    if margins:
+        _check_margins_name(margins_name, table)
+
+        if normalize != 'index':
+            # add margin column
+            table[margins_name] = table.sum(axis=1)
+
+        if normalize != 'columns':
+            # add margin row
+            if type(table.index) is MultiIndex:
+                table = table.transpose()
+                table[margins_name] = table.sum(axis=1)
```

Comment:
I have a solution now:

    df.loc[margins_name, :] = df.sum(axis=1)

I'll make a new commit soon.

see here
https://stackoverflow.com/questions/44949953/how-to-add-a-row-to-a-pandas-dataframe-without-flattening-the-multiindex